### PR TITLE
[dashboard] Remove redis in dashboard

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
As we are turning redisless ray by default, dashboard doesn't need to talk with redis anymore. Instead it should talk with gcs and gcs can talk with redis.

Clean up the code.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
The next step is to remove ray core - python redis related code. And then I'll remove redis from `setup.py`

File: python/setup.py [hunk 1]
```diff
@@ -146,7 +146,6 @@ def get_packages(self):
 
 # NOTE: The lists below must be kept in sync with ray/BUILD.bazel.
 ray_files = [
-    "ray/core/src/ray/thirdparty/redis/src/redis-server" + exe_suffix,
```
Comment (ericl) [inline python/setup.py:149]: Safe to remove?

**Review by ericl — APPROVED** [review]
LGTM! Btw, should we remove our "GCS HA" buildkite builds? Or should those test the "with Redis" path?

File: dashboard/head.py [hunk 2]
```diff
@@ -220,24 +189,7 @@ def _load_modules(self):
         return modules
 
     async def get_gcs_address(self):
```
Comment (mwtian) [inline dashboard/head.py:191]: Remove async keyword, or migrate callers to self.gcs_address?

File: dashboard/modules/reporter/reporter_agent.py [hunk 3]
```diff
@@ -610,28 +604,13 @@ async def _perform_iteration(self, publish):
...
-                address=self._dashboard_agent.redis_address,
-                password=self._dashboard_agent.redis_password,
-            )
+        gcs_addr = self._dashboard_agent.gcs_address
+        assert gcs_addr is not None
+        publisher = GcsAioPublisher(address=gcs_addr)
 
-            async def publish(key: str, data: str):
-                await aioredis_client.publish(key, data)
+        async def publish(key: str, data: str):
```
Comment (mwtian) [inline dashboard/modules/reporter/reporter_agent.py:612]: Maybe this can be merged into `_perform_iteration()`, instead of having to pass into it.

**Review by mwtian — APPROVED** [review]

File: python/setup.py (hunk 1 above)
Comment (iycheng) [reply python/setup.py:149]: Sorry, my bad. I'll put it back... it's left from the previous PR.

**Comment by iycheng** [comment]
> LGTM! Btw, should we remove our "GCS HA" buildkite builds? Or should those test the "with Redis" path?

I think redis path has some issues right now because we didn't test that :( and we'll need to fix it and add with redis path.

So in the end, we'll test memory storage and redis storage code path.

**Comment by iycheng** [comment]
We'll fix that once booststrap and kv redis code paths are deleted. 
@mwtian will work on delete pubsub redis code path. But I think it's won't impact the fixing of redis storage code path.

**Pull request merged**